Jenkinsfile

```diff
@@ -369,28 +396,81 @@ pipeline {
 
     stage('Run runtime unit tests') {
       steps {
-        setGitHubBuildStatus('platform/utests/runtime/dev', 'Unit tests - runtime', 'PENDING')
         container('maven') {
-          echo """
-          ----------------------------------------
-          Run runtime unit tests
-          ----------------------------------------"""
-          dir('modules/runtime') {
-            sh "mvn ${MAVEN_ARGS} test"
+          script {
+            setGitHubBuildStatus('platform/utests/runtime', 'Unit tests - runtime', 'PENDING')
+            def testNamespace = "${TEST_NAMESPACE_PREFIX}-runtime"
+            def redisHost = "${TEST_REDIS_RESOURCE}.${testNamespace}.${TEST_SERVICE_DOMAIN_SUFFIX}"
+            def kafkaHost = "${TEST_KAFKA_RESOURCE}.${testNamespace}.${TEST_SERVICE_DOMAIN_SUFFIX}:${TEST_KAFKA_PORT}"
+            try {
+              echo """
+              ----------------------------------------
+              Run runtime unit tests
+              ----------------------------------------"""
+
+              echo "runtime unit tests: install external services"
+              // initialize Helm without Tiller and add local repository
+              sh """
+                helm init --client-only
+                helm repo add ${HELM_CHART_REPOSITORY_NAME} ${HELM_CHART_REPOSITORY_URL}
+              """
+              // install the nuxeo Helm chart into a dedicated namespace that will be cleaned up afterwards
+              sh """
+                jx step helm install ${HELM_CHART_REPOSITORY_NAME}/${HELM_CHART_NUXEO} \
+                  --name=${TEST_HELM_CHART_RELEASE} \
+                  --namespace=${testNamespace} \
+                  --set-file=ci/helm/nuxeo-test-base-values.yaml~gen \
+                  --set-file=ci/helm/nuxeo-test-kafka-values.yaml~gen
+              """
+              // wait for Redis to be ready
+              sh """
+                kubectl rollout status statefulset ${TEST_REDIS_RESOURCE} \
+                  --namespace=${testNamespace} \
+                  --timeout=${TEST_DEFAULT_ROLLOUT_STATUS_TIMEOUT}
+              """
+              // wait for Kafka to be ready
+              sh """
+                kubectl rollout status statefulset ${TEST_KAFKA_RESOURCE} \
+                  --namespace=${testNamespace} \
+                  --timeout=${TEST_LONG_ROLLOUT_STATUS_TIMEOUT}
+              """
```

Comment:
Could it be interesting to have a method for the rollouts? something like waitFor(ns, type, name, timeout)? Or without type as we know all kinds of resources and can do a switch in the method, wdyt?